js/components/operatorSignIn/expiry.tsx

```diff
@@ -0,0 +1,163 @@
+import { daysBetween } from "../../dateTime";
+import {
+  Certification,
+  humanReadableType,
+  isExpired,
+} from "../../models/certification";
+import { className } from "../../util/dom";
+import { WarningParagraph } from "./warningParagraph";
+import { DateTime } from "luxon";
+import { ReactElement } from "react";
+
+const WARN_WITHIN_D = 60;
+
+const englishDaysBetween = (now: DateTime, date: DateTime) => {
+  const delta = daysBetween(now, date);
+  const wholeAbs = Math.abs(Math.ceil(delta));
+  if (delta <= -2) {
+    return `expired ${wholeAbs} days ago`;
+  } else if (delta <= -1) {
+    return "expired yesterday";
+  } else if (delta <= 0) {
+    return "expired today";
+  } else if (delta <= 1) {
+    return "expires tomorrow";
+  } else {
+    return `expires in ${wholeAbs} days`;
+  }
+};
+
+const CertificateBox = ({
+  now,
+  mode,
+  title,
+  operatorName,
+  certifications,
+}: {
+  now: DateTime;
+  mode: "warning" | "error";
+  title: string;
+  operatorName: string;
+  certifications: Certification[];
+}): ReactElement => {
+  const innerString =
+    `Our records show that ${operatorName}'s ` +
+    certifications
+      .map(
+        (c) =>
+          `${humanReadableType(c.type)} ${englishDaysBetween(
+            now,
+            c.expires,
+          )} on ${c.expires.toLocaleString(DateTime.DATE_SHORT)}`,
+      )
+      .join(" and ") +
+    ".";
+  return (
+    <WarningParagraph
+      className={className([
+        "border-0 light:bg-opacity-40 dark:bg-opacity-30 dark:text-white",
+        mode === "warning" && "bg-[#FFDE9E]",
+        mode === "error" && "bg-[#FF919A]",
+      ])}
+    >
+      <p className="font-bold uppercase">{title}</p>
+      <p className="mt-2">{innerString}</p>
+      {mode === "warning" && (
+        <p className="mt-2">Please have them call the Office.</p>
+      )}
+    </WarningParagraph>
+  );
+};
+
+export const CertificateBoxes = ({
+  now,
+  displayName,
+  ignoreExpired,
+  certifications,
+}: {
+  now: DateTime;
+  displayName: string;
+  ignoreExpired: boolean;
+  certifications: Certification[];
+}): ReactElement => {
+  const expired = certifications.filter((cert) => isExpired(cert, now));
+  const expiresSoon = certifications.filter((cert) => {
+    const delta = daysBetween(now, cert.expires);
+    return !isExpired(cert, now) && delta <= WARN_WITHIN_D;
+  });
+  return (
+    <>
+      {expiresSoon.length > 0 && (
+        <CertificateBox
+          now={now}
+          mode="warning"
+          title={
+            expiresSoon.length === 2 ? "Cards expire soon" : "Card expires soon"
+          }
+          operatorName={displayName}
+          certifications={expiresSoon}
+        />
+      )}
+      {expired.length > 0 && !ignoreExpired && (
+        <CertificateBox
+          now={now}
+          mode="error"
+          title={expired.length === 2 ? "Expired cards" : "Expired card"}
```

Comment:
nitpick: This is fragile if anybody ever has more than 2 certifications (could happen if we track more than two types, or if someone has certs on more than one rail line). It should be `expired.length === 1 ? singular : plural` or `expired.length > 1 ? plural : singular`.

Also applies to the previous call 10 lines above this.

Reply:
fixed 👍 thanks